[JDK 17] [Native-image] Random/SplittableRandom class in the image heap, reading field sun.security.jca.JCAUtil.def

### Describe the bug

Compiling an application with JDK 17 based [Mandrel 21.3-snapshot](https://ci.modcluster.io/view/Mandrel/job/mandrel-master-jdk17-linux-build/2/artifact/mandrel-java17-linux-amd64-21.3.0-deve98ae39603.tar.gz) distribution fails with the infamous:

```
[INFO] [io.quarkus.deployment.pkg.steps.NativeImageBuildStep] Running Quarkus native-image plugin on native-image 21.3.0-deve98ae39603 Mandrel Distribution (Java Version 17-internal+0-adhoc.tester.jdk17)
[INFO] [io.quarkus.deployment.pkg.steps.NativeImageBuildRunner] /home/karm/workspaceRH/mandrel-release/mandrel-java17-21.3.0-deve98ae39603/bin/native-image -J-Dsun.nio.ch.maxUpdateArraySize=100 -J-Djava.util.logging.manager=org.jboss.logmanager.LogManager -J-Dvertx.logger-delegate-factory-class-name=io.quarkus.vertx.core.runtime.VertxLogDelegateFactory -J-Dvertx.disableDnsResolver=true -J-Dio.netty.leakDetection.level=DISABLED -J-Dio.netty.allocator.maxOrder=3 -J-Duser.language=en -J-Duser.country=US -J-Dfile.encoding=UTF-8 -H:Log=registerResource: -H:IncludeResources=privateKey.pem --trace-object-instantiation=java.util.Random --initialize-at-run-time=io.vertx.ext.auth.impl.jose.JWT -H:InitialCollectionPolicy=com.oracle.svm.core.genscavenge.CollectionPolicy\$BySpaceAndTime -H:+JNI -H:+AllowFoldMethods -jar quarkus-runner.jar -H:FallbackThreshold=0 -H:+ReportExceptionStackTraces -H:-AddAllCharsets -H:EnableURLProtocols=http,https -H:NativeLinkerOption=-no-pie -H:-UseServiceLoaderFeature -H:+StackTrace -H:-ParseOnce quarkus-runner
[quarkus-runner:199424]    classlist:   1,864.62 ms,  0.94 GB
[quarkus-runner:199424]        (cap):     506.28 ms,  0.94 GB
[quarkus-runner:199424]        setup:   1,983.65 ms,  0.94 GB
The bundle named: messages, has not been found. If the bundle is part of a module, verify the bundle name is a fully qualified class name. Otherwise verify the bundle path is accessible in the classpath.
00:15:11,833 INFO  [org.jbo.res.res.i18n] RESTEASY002225: Deploying javax.ws.rs.core.Application: class com.example.quarkus.QuarkusRestApplication
00:15:27,453 INFO  [org.jbo.threads] JBoss Threads version 3.4.2.Final
[quarkus-runner:199424]     (clinit):     626.98 ms,  5.59 GB
[quarkus-runner:199424]   (typeflow):   2,880.06 ms,  5.59 GB
[quarkus-runner:199424]    (objects):  33,213.96 ms,  5.59 GB
[quarkus-runner:199424]   (features):   1,825.25 ms,  5.59 GB
[quarkus-runner:199424]     analysis:  40,037.04 ms,  5.59 GB
Error: Detected an instance of Random/SplittableRandom class in the image heap. Instances created during image generation have cached seed values and don't behave as expected.  To see how this object got instantiated use --trace-object-instantiation=java.security.SecureRandom. The object was probably created by a class initializer and is reachable from a static field. You can request class initialization at image runtime by using the option --initialize-at-run-time=<class-name>. Or you can write your own initialization methods and call them explicitly from your main entry point.
Detailed message:
Trace: Object was reached by 
	reading field sun.security.jca.JCAUtil.def

com.oracle.svm.core.util.UserError$UserException: Detected an instance of Random/SplittableRandom class in the image heap. Instances created during image generation have cached seed values and don't behave as expected.  To see how this object got instantiated use --trace-object-instantiation=java.security.SecureRandom. The object was probably created by a class initializer and is reachable from a static field. You can request class initialization at image runtime by using the option --initialize-at-run-time=<class-name>. Or you can write your own initialization methods and call them explicitly from your main entry point.
Detailed message:
Trace: Object was reached by 
	reading field sun.security.jca.JCAUtil.def

	at com.oracle.svm.core.util.UserError.abort(UserError.java:82)
	at com.oracle.svm.hosted.FallbackFeature.reportAsFallback(FallbackFeature.java:233)
	at com.oracle.svm.hosted.NativeImageGenerator.runPointsToAnalysis(NativeImageGenerator.java:765)
	at com.oracle.svm.hosted.NativeImageGenerator.doRun(NativeImageGenerator.java:533)
	at com.oracle.svm.hosted.NativeImageGenerator.run(NativeImageGenerator.java:492)
	at com.oracle.svm.hosted.NativeImageGeneratorRunner.buildImage(NativeImageGeneratorRunner.java:400)
	at com.oracle.svm.hosted.NativeImageGeneratorRunner.build(NativeImageGeneratorRunner.java:566)
	at com.oracle.svm.hosted.NativeImageGeneratorRunner.main(NativeImageGeneratorRunner.java:122)
	at com.oracle.svm.hosted.NativeImageGeneratorRunner$JDK9Plus.main(NativeImageGeneratorRunner.java:596)
Caused by: com.oracle.graal.pointsto.constraints.UnsupportedFeatureException: Detected an instance of Random/SplittableRandom class in the image heap. Instances created during image generation have cached seed values and don't behave as expected.  To see how this object got instantiated use --trace-object-instantiation=java.security.SecureRandom. The object was probably created by a class initializer and is reachable from a static field. You can request class initialization at image runtime by using the option --initialize-at-run-time=<class-name>. Or you can write your own initialization methods and call them explicitly from your main entry point.
Detailed message:
Trace: Object was reached by 
	reading field sun.security.jca.JCAUtil.def

	at com.oracle.graal.pointsto.constraints.UnsupportedFeatures.report(UnsupportedFeatures.java:126)
	at com.oracle.svm.hosted.NativeImageGenerator.runPointsToAnalysis(NativeImageGenerator.java:762)
	... 6 more
[quarkus-runner:199424]      [total]:  44,141.34 ms,  5.59 GB
```

### Expected behavior

Just as with JDK 11 based [Mandrel 21.3-snapshot](https://ci.modcluster.io/view/Mandrel/job/mandrel-master-linux-build/lastSuccessfulBuild/artifact/mandrel-java11-linux-amd64-21.3.0-deve98ae39603.tar.gz), the build should pass:

```
[INFO] [io.quarkus.deployment.pkg.steps.NativeImageBuildStep] Running Quarkus native-image plugin on native-image 21.3.0-dev5aa724e48260 Mandrel Distribution (Java Version 11.0.12+7)
[INFO] [io.quarkus.deployment.pkg.steps.NativeImageBuildRunner] /home/karm/tmp/mandrel-java11-21.3-SNAPSHOT/bin/native-image -J-Dsun.nio.ch.maxUpdateArraySize=100 -J-Djava.util.logging.manager=org.jboss.logmanager.LogManager -J-Dvertx.logger-delegate-factory-class-name=io.quarkus.vertx.core.runtime.VertxLogDelegateFactory -J-Dvertx.disableDnsResolver=true -J-Dio.netty.leakDetection.level=DISABLED -J-Dio.netty.allocator.maxOrder=3 -J-Duser.language=en -J-Duser.country=US -J-Dfile.encoding=UTF-8 -H:Log=registerResource: -H:IncludeResources=privateKey.pem --trace-object-instantiation=java.util.Random --initialize-at-run-time=io.vertx.ext.auth.impl.jose.JWT -H:InitialCollectionPolicy=com.oracle.svm.core.genscavenge.CollectionPolicy\$BySpaceAndTime -H:+JNI -H:+AllowFoldMethods -jar quarkus-runner.jar -H:FallbackThreshold=0 -H:+ReportExceptionStackTraces -H:-AddAllCharsets -H:EnableURLProtocols=http,https -H:NativeLinkerOption=-no-pie -H:-UseServiceLoaderFeature -H:+StackTrace -H:-ParseOnce quarkus-runner
[quarkus-runner:198124]    classlist:   2,268.02 ms,  1.18 GB
[quarkus-runner:198124]        (cap):     479.31 ms,  1.18 GB
[quarkus-runner:198124]        setup:   2,205.93 ms,  1.18 GB
The bundle named: messages, has not been found. If the bundle is part of a module, verify the bundle name is a fully qualified class name. Otherwise verify the bundle path is accessible in the classpath.
00:06:19,566 INFO  [org.jbo.res.res.i18n] RESTEASY002225: Deploying javax.ws.rs.core.Application: class com.example.quarkus.QuarkusRestApplication
00:06:34,784 INFO  [org.jbo.threads] JBoss Threads version 3.4.2.Final
[quarkus-runner:198124]     (clinit):     805.25 ms,  5.51 GB
[quarkus-runner:198124]   (typeflow):   2,811.71 ms,  5.51 GB
[quarkus-runner:198124]    (objects):  35,825.59 ms,  5.51 GB
[quarkus-runner:198124]   (features):   1,152.99 ms,  5.51 GB
[quarkus-runner:198124]     analysis:  42,050.04 ms,  5.51 GB
[quarkus-runner:198124]     universe:   3,050.45 ms,  5.51 GB
[quarkus-runner:198124]      (parse):   8,898.52 ms,  5.91 GB
[quarkus-runner:198124]     (inline):   4,469.41 ms,  5.66 GB
[quarkus-runner:198124]    (compile):  32,873.37 ms,  6.77 GB
[quarkus-runner:198124]      compile:  48,812.73 ms,  6.77 GB
[quarkus-runner:198124]        image:   4,037.39 ms,  6.90 GB
[quarkus-runner:198124]        write:     777.24 ms,  6.90 GB
[quarkus-runner:198124]      [total]: 103,563.36 ms,  6.90 GB
# Printing build artifacts to: /home/karm/workspaceRH/mandrel-integration-tests/apps/quarkus-full-microprofile/target/quarkus-native-image-source-jar/quarkus-runner.build_artifacts.txt
[INFO] [io.quarkus.deployment.pkg.steps.NativeImageBuildRunner] objcopy --strip-debug quarkus-runner
[INFO] [io.quarkus.deployment.QuarkusAugmentor] Quarkus augmentation completed in 106972ms

2021-08-25 00:12:22,968 INFO  [io.quarkus] (main) quarkus-full-microprofile 1.0.0-SNAPSHOT native (powered by Quarkus 999-SNAPSHOT) started in 0.018s. Listening on: http://0.0.0.0:8080
2021-08-25 00:12:22,969 INFO  [io.quarkus] (main) Profile prod activated. 
2021-08-25 00:12:22,969 INFO  [io.quarkus] (main) Installed features: [cdi, jaeger, rest-client, resteasy, security, smallrye-context-propagation, smallrye-fault-tolerance, smallrye-health, smallrye-jwt, smallrye-metrics, smallrye-openapi, smallrye-opentracing, vertx, vertx-web]
```

### Actual behavior

Missing substitution?

### How to Reproduce?

```
wget https://ci.modcluster.io/view/Mandrel/job/mandrel-master-jdk17-linux-build/2/artifact/mandrel-java17-linux-amd64-21.3.0-deve98ae39603.tar.gz
tar -xf mandrel-java17-linux-amd64-21.3.0-deve98ae39603.tar.gz
export JAVA_HOME=$(pwd)/mandrel-java17-21.3.0-deve98ae39603;export GRAALVM_HOME=${JAVA_HOME};export PATH=${JAVA_HOME}/bin:${PATH}
git clone https://github.com/Karm/mandrel-integration-tests
pushd mandrel-integration-tests/apps/quarkus-full-microprofile/
mvn clean compile package -Pnative -Dquarkus.version=999-SNAPSHOT
```

### Output of `uname -a` or `ver`

Linux  4.18.0-240.22.1.el8_3.x86_64

### Output of `java -version`

JDK 17u, git describe: jdk-17+35-37-g648f3f6a8fe

### GraalVM version (if different from Java)

native-image 21.3.0-deve98ae39603 Mandrel Distribution (Java Version 17-internal+0-adhoc.tester.jdk17)

### Quarkus version or git rev

main, 0110329564

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.6.3

### Additional information

_No response_

--- a/core/runtime/src/main/java/io/quarkus/runtime/graal/Target_sun_security_jca_JCAUtil.java
+++ b/core/runtime/src/main/java/io/quarkus/runtime/graal/Target_sun_security_jca_JCAUtil.java
@@ -1,0 +1,16 @@
+package io.quarkus.runtime.graal;
+
+import java.security.SecureRandom;
+
+import com.oracle.svm.core.annotate.Alias;
+import com.oracle.svm.core.annotate.RecomputeFieldValue;
+import com.oracle.svm.core.annotate.TargetClass;
+import com.oracle.svm.core.jdk.JDK17OrLater;
+
+@TargetClass(className = "sun.security.jca.JCAUtil", onlyWith = JDK17OrLater.class)
+public final class Target_sun_security_jca_JCAUtil {
+
+    @Alias
+    @RecomputeFieldValue(kind = RecomputeFieldValue.Kind.Reset)
+    private static SecureRandom def;
+}

